### ISSUE TITLE
Call to a member function getConfiguration() on null

### ISSUE THREAD

**Issue by nbiwki** [dmind-gmbh/extension-cookieman#119]
# Bug Report

## Prerequisites

* [ ] Can you reproduce the problem on TYPO3 v8.7 LTS
* [ ] Can you reproduce the problem on TYPO3 v9.5 LTS
* [ ] Can you reproduce the problem on TYPO3 v10.4 LTS
* [ ] Can you reproduce the problem on TYPO3 master

I can't reproduce the error, just during an installation. This applies to TYPO3 8.7 with PHP 7.0.

## Description

As soon as I activate the static templates I get the error message

Uncaught TYPO3 Exception: Call to a member function getConfiguration() on null | Error thrown in file /html/example.de/vendor/typo3/cms/typo3/sysext/extbase/Classes/Configuration/ConfigurationManager.php in line 124. Requested URL: https://www.example.de/de/

For other TYPO3 installations with the same versions, this works without any problems.

## Versions

TYPO3 8.7.28 composer based.

`algo26-matthias/idna-convert        v1.1.0                                A library for encoding and decoding internationalized domain names
clickstorm/cs_seo                   2.3.4                                 [clickstorm] SEO
cogpowered/finediff                 0.3.1                                 PHP implementation of a Fine granularity Diff engine
consolidation/annotated-command     2.12.0                                Initialize Symfony Console commands from annotated command class methods.
consolidation/config                1.2.1                                 Provide configuration services for a commandline tool.
consolidation/log                   1.1.1                                 Improved Psr-3 / Psr\Log logger based on Symfony Console components.
consolidation/output-formatters     3.5.0                                 Format text by applying transformations provided by plug-in formatters.
consolidation/robo                  1.4.10                                Modern task runner
consolidation/self-update           1.1.5                                 Provides a self:update command for Symfony Console applications.
container-interop/container-interop 1.2.0                                 Promoting the interoperability of container objects (DIC, SL, etc.)
dflydev/dot-access-data             v1.1.0                                Given a deep data structure, access data by dot notation.
dmind/cookieman                     2.4.1                                 A GDPR tracking consent popup. It asks for approval to include tracking objects (cookies, images or any HTML) a...
dmitryd/typo3-realurl               2.5.0                                 Speaking URLs for TYPO3
doctrine/annotations                v1.4.0                                Docblock Annotations Parser
doctrine/cache                      v1.6.2                                Caching library offering an object-oriented API for many cache backends
doctrine/collections                v1.4.0                                Collections Abstraction library
doctrine/common                     v2.7.3                                Common Library for Doctrine projects
doctrine/dbal                       v2.5.13                               Database Abstraction Layer
doctrine/inflector                  v1.2.0                                Common String Manipulations with regard to casing and singular/plural rules.
doctrine/instantiator               1.0.5                                 A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                      1.0.2                                 PHP Doctrine Lexer parser library that can be used in Top-Down, Recursive Descent Parsers.
fluidtypo3/fluidcontent             5.2.0                                 Create Flexible Content elements in pure fluid
fluidtypo3/fluidpages               4.1.0                                 The fluidpages package from FluidTYPO3
fluidtypo3/flux                     8.2.1                                 The flux package from FluidTYPO3
fluidtypo3/vhs                      4.4.0                                 The vhs package from FluidTYPO3
georgringer/news                    6.3.0                                 Versatile news extension, based on extbase & fluid for TYPO3 CMS
grasmash/expander                   1.0.0                                 Expands internal property references in PHP arrays file.
grasmash/yaml-expander              1.4.0                                 Expands internal property references in a yaml file.
guzzlehttp/guzzle                   6.3.3                                 Guzzle is a PHP HTTP client library
guzzlehttp/promises                 v1.3.1                                Guzzle promises library
guzzlehttp/psr7                     1.6.1                                 PSR-7 message implementation that also provides common utility methods
helhum/config-loader                v0.12.2                               Generic config loader with context and environment support.
helhum/typo3-console                v5.7.2                                A reliable and powerful command line interface for TYPO3 CMS
helhum/typo3-console-plugin         v2.0.5                                Installer plugin for helhum/typo3-console, to ease usage without the need to specify scripts in your root compo...
koninklijke-collective/my-redirects 3.4.1                                 TYPO3 Extension: Redirects management
league/container                    2.4.1                                 A fast and intuitive dependency injection container.
mindshape/mindshape_cookie_hint     1.0.14                                Shows a visible hint on the website that cookies are used - according to the European laws. Easy to install, fu...
nordcode/robo-bump                  dev-master 19eb897                    Version bump task for robo task runner
nwsnet/fluidbootstraptheme          dev-feature/8-7-compatibility 34a1905 fluidbootstraptheme for TYPO3 8.7
nwsnet/nws-scaffolding              3.0.2                                 TYPO3 extension that provides common templates and TypoScript configuration for TYPO3 or extensions
phorax/formhandler                  3.1.0                                 The swiss army knife for all kinds of mailforms, completely new written using the MVC concept. Result: Flexibil...
psr/container                       1.0.0                                 Common Container Interface (PHP FIG PSR-11)
psr/http-message                    1.0.1                                 Common interface for HTTP messages
psr/log                             1.0.2                                 Common interface for logging libraries
ralouphie/getallheaders             3.0.3                                 A polyfill for getallheaders.
swiftmailer/swiftmailer             v5.4.12                               Swiftmailer, free feature-rich PHP mailer
symfony/console                     v3.4.32                               Symfony Console Component
symfony/debug                       v3.4.32                               Symfony Debug Component
symfony/event-dispatcher            v3.4.32                               Symfony EventDispatcher Component
symfony/filesystem                  v3.4.32                               Symfony Filesystem Component
symfony/finder                      v3.4.32                               Symfony Finder Component
symfony/polyfill-ctype              v1.12.0                               Symfony polyfill for ctype functions
symfony/polyfill-mbstring           v1.12.0                               Symfony polyfill for the Mbstring extension
symfony/process                     v3.4.32                               Symfony Process Component
symfony/yaml                        v3.4.32                               Symfony Yaml Component
t3monitor/t3monitoring_client       dev-7-8 c5f10b4                       Client extension for the t3monitoring service
typo3-ter/compatibility7            8.7.1                                 Provides an additional backwards-compatibility layer with legacy functionality for sites that havent fully migr...
typo3-ter/static-info-tables        6.7.5                                 Data and API for countries, languages and currencies.
typo3/class-alias-loader            1.0.1                                 Amends the composer class loader to support class aliases to provide backwards compatibility for packages
typo3/cms                           v8.7.28                               TYPO3 CMS is a free open source Content Management Framework initially created by Kasper Skaarhoj and licensed ...
typo3/cms-cli                       1.0.2                                 TYPO3 command line binary
typo3/cms-composer-installers       v1.5.4                                TYPO3 CMS Installers
typo3/phar-stream-wrapper           v3.1.3                                Interceptors for PHP's native phar:// stream handling
typo3fluid/fluid                    2.6.4                                 The TYPO3 Fluid template rendering engin`

**Comment by jonaseberledmind** [comment]
Sorry, I can't follow. 
You mean you have an error during composer installation? Or after you save in the template module when including the static templates? 
But no error afterwards? 

Looking at https://github.com/TYPO3/TYPO3.CMS/blob/TYPO3_8-7/typo3/sysext/extbase/Classes/Configuration/ConfigurationManager.php#L124 this seems to come from deep inside Extbase...

**Comment by jonaseberledmind** [comment]
Also the line number does not match your error. You might want to update to the 8LTS version released yesterday.

**Comment by nbiwki** [comment]
the composer installation works well.

After i save the Template with the TypoScript won't work.

I found the error now, it is in the file Classes/DataProcessing/TypoScriptSettingsProcessor.php 

The getConfigurationManager() function incorrectly calls the ConfigurationManager.

Instead of GeneralUtility::makeInstance(ConfigurationManager::class), it must be done this way:

```
$objectManager = GeneralUtility::makeInstance('TYPO3\\CMS\Extbase\\Object\\ObjectManager');
$objectManager->get(ConfigurationManager::class);
```

GeneralUtility::makeInstance only works like a new-operator, but objectManager also takes care of all the TYPO3 relevant parts such as "@inject", which ensures that the Object ConfigurationManager contains all the necessary objects.

Why does it still work for most? Quite simply, the ConfigurationManager class is a singleton, and GeneralUtility::makeInstance respects that. If another plugin has previously initialized ConfigurationManager correctly, it will also work for Cookieman. However, if Cookieman makes the first call to ConfigurationManager, the error comes.

**Comment by jonaseberledmind** [comment]
You are right. As this is an extbasey Service, we must use the Extbase ObjectManager.
Maybe it makes sense to move away from Extbase anyways to get TypoScript. I'll have a look.

For 10 (master) we can use DI which we should use now as well. 

Would you like to do a PR for 8 and 9 with your code? I would be very thankful.
